### PR TITLE
Refactoring Case Entity to ES6 Class

### PR DESCRIPTION
**Pull request by codyseibert**

### CODE REVIEW

**Review by zachrog — APPROVED** [review]

File: shared/src/business/entities/cases/Case.ts [hunk 1]
```diff
@@ -24,2501 +23,2441 @@ const {
...
+    this.orderForCds = rawCase.orderForCds || false;
+    this.orderForRatification = rawCase.orderForRatification || false;
+    this.orderToShowCause = rawCase.orderToShowCause || false;
+
+    this.assignArchivedDocketEntries({ applicationContext, rawCase });
+    this.assignStatistics({ applicationContext, rawCase });
+    this.assignCorrespondences({ rawCase });
+  }
+
+  static VALIDATION_RULES = {
```
Comment (Absolutestunna) [inline shared/src/business/entities/cases/Case.ts:383]: Hey @codyseibert is this better housed in the entityvalidationConstants file?

**Review by ujahio — APPROVED** [review]

File: shared/src/business/entities/cases/Case.ts (hunk 1 above)
Comment (ujahio) [inline shared/src/business/entities/cases/Case.ts:383]: Hey @codyseibert are the case validation rules better housed in the entityvalidationConstants file?

Comment (codyseibert) [reply shared/src/business/entities/cases/Case.ts:383]: I'm not sure why the validation rules for entities are in that file to begin with.  I would think validation logic related to the entity should live in the file that creates the entity.

Comment (zachrog) [reply shared/src/business/entities/cases/Case.ts:383]: I think there are some Validation rules which are shared between entities. Such as startDate, endDate validation rules could be reused by many entities and is common so gets extracted to entityValidationConstants. However if the validation rules are not shared anywhere then it should probably just live in the entity itself

**Review by shumway-tm — APPROVED** [review]

**Review by rgoshen — APPROVED** [review]

**Review by mmarcotte — APPROVED** [review]

**Pull request merged**